internal/provider/deployment.go

```diff
@@ -364,7 +368,7 @@ func (d *DeploymentDeployer) validate() error {
 	}
 	return nil
 }
-func (d *DeploymentDeployer) Deploy(ctx context.Context, sub subi.SubstrateExt) error {
+func (d *DeploymentDeployer) Deploy(ctx context.Context, sub subi.Substrate) error {
 	if err := d.validate(); err != nil {
 		return err
 	}
```

Comment:
for error in line 377 it's single deployment and maybe use the deployment id too